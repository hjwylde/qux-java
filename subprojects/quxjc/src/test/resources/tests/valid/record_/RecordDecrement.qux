package valid.record_

{int x, int y} f({int x, int y} r):
    r.x--

    return r

void main([str] args):
    xs = {x: 0, y: 1, z: {x: 2}}
    print xs.x
    print xs.y
    print xs.z
    xs.x--
    xs.y--
    xs.z.x--
    print xs.x--
    print xs.x
    print xs.y
    print xs.z
    print f(xs)
    print xs
