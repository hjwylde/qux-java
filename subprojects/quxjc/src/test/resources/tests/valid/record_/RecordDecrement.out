0
1
{x: 2}
-1
-2
0
{x: 1}
{x: -3, y: 0, z: {x: 1}}
{x: -2, y: 0, z: {x: 1}}
